python_modules/dagster/dagster_tests/definitions_tests/decorators_tests/test_source_asset.py

```diff
@@ -33,3 +38,31 @@ def foo_source_asset(context):
     assert foo_source_asset.resource_defs == {"lambda": foo_manager}
     assert foo_source_asset.io_manager_def == foo_manager
     assert foo_source_asset.metadata == {"epsilon": MetadataValue.text("gamma")}
+
+
+def test_no_context_observable_asset():
+    executed = {}
+
+    @observable_source_asset
+    def observable_asset_no_context():
+        executed["yes"] = True
+        return LogicalVersion("version-string")
+
+    # does not work
+    # asset_job = define_asset_job(name="source_assets", selection="*")
+
+    # works
+    # asset_job = build_assets_job(name="source_job", assets=[], source_assets=[observable_asset_no_context])
```

Comment:
@sryza @smackesey This was particularly confusing. Is there a way to execute this using `define_asset_job`. Both `build_assets_job` and `build_source_asset_observation_job` are experimental.

Reply:
But am I correct that there is currently _no_ way to use `define_asset_job` to include an observable source asset? Is there a way to invoke an observable source asset without using an experimental API?

Reply:
After playing with this, this is correct:

>there is currently no way to use define_asset_job to include an observable source asset

It wasn't always this way-- I think recent changes have caused this. It's my bad for lacking test coverage here, and I'm fixing it right now.